syntax = "proto3";

package cn.mx404.audiotoass;

service AudioStream {
    rpc GetConfig(SDKVersion) returns (Config) {}

    rpc GetStream(Empty) returns (stream Frame) {}

    rpc RecognitionStart(Empty) returns(Empty) {}
    rpc RecognitionEnd(Empty) returns(Empty) {}
    rpc SentenceEnd(Empty) returns(Empty) {}
    rpc SentenceStart(Empty) returns(Empty) {}

    rpc SentenceFinishData(JsonString) returns(Empty) {}
    rpc SentenceFlushData(JsonString) returns(Empty) {}

    rpc SDKError(ErrorMessage) returns(Empty) {}
    rpc SDKUnknowError(ErrorContent) returns(Empty) {}

    rpc SDKStartFail(ErrorString) returns(Empty) {}
    rpc WrapMessageFail(ErrorString) returns(Empty) {}
}

message Empty {
}

message SDKVersion {
    string version = 1;
}

message Frame {
    bytes data = 1;
    bool cancel = 2;
}

message ErrorMessage {
    int32 code = 1;
    int32 domain = 2;
    string description = 3;
    string serialNumber = 4;
}

message JsonString {
    string data = 1;
}

message Config {
    repeated string unknowErrorKeys = 1;
    string appID = 2;
    string appName = 3;
    string chunkKey = 4;
    int32 productID = 5;
    string dnnDatFilePath = 6;
    string cmvnDatFilePath = 7;
}

message ErrorContent {
    int32 statusCode = 1;
    map<string, string> data = 2;
}

message ErrorString {
    string errorString = 1;
}
